specification/dartLangSpec.tex

```diff
@@ -16276,38 +16279,40 @@ \subsection{Assignable Expressions}
 \end{grammar}
 
 \LMHash{}%
-An \Index{assignable expression} is either:
-\begin{itemize}
-\item An identifier.
-\item An invocation (possibly conditional) of a getter (\ref{getters})
-  or list access operator on an expression $e$.
-\item An invocation of a getter or list access operator on \SUPER.
-\end{itemize}
-
-\LMHash{}%
-An assignable expression of the form \id{} is evaluated as
-an identifier expression (\ref{identifierReference}).
-
-%An assignable expression of the form \code{$e$.\id($a_1, \ldots,\ a_n$)}
-% is evaluated as a method invocation (\ref{methodInvocation}).
-
-\LMHash{}%
-An assignable expression of the form \code{$e$.\id} or \code{$e$?.\id}
-is evaluated as a property extraction (\ref{propertyExtraction}).
-
-\LMHash{}%
-An assignable expression of the form \code{$e_1$[$e_2$]}
-is evaluated as a method invocation of
-the operator method \code{[]} on $e_1$ with argument $e_2$.
-
-\LMHash{}%
-An assignable expression of the form \code{\SUPER.\id} is evaluated as
-a property extraction.
+\BlindDefineSymbol{e, t}%
+Let $e$ be an \synt{assignableExpression}.
+Let $t$ be a term such that $e$ can be derived from
+\syntax{$t$ <assignableSelector>}.
+We say that $t$ is the
+\IndexCustom{receiver term}{assignable expression!receiver term}
+of $e$.
+When $t$ is an expression, we say that $t$ is the
+\IndexCustom{receiver expression}{assignable expression!receiver expression}
```

Comment:
(Can't see it in this change, so just asking instead of trying to find the text:
Do we include `prefix` as a receiver expression? That is, do we differentiate between receiver terms which are grammatically an expression or not, or between receiver terms which are *semantically* evaluable expressions or not? Should we? It might make things easier in other places.)

Reply:
The text now states explicitly that the cases in section Assignment are applicable when the side conditions are satisfied (e.g., `C.v = e` is applicable when `C` is a type literal), and that the forms containing expressions (such as `e1.v = e2`) are only applicable when the more specific forms do not apply.